packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -190,26 +190,14 @@ export class RunningSummarizer implements IDisposable {
             ? this.logger
             : undefined;
 
-    public handleSystemOp(op: ISequencedDocumentMessage) {
-        switch (op.type) {
-            case MessageType.ClientLeave:
-            case MessageType.ClientJoin:
-            case MessageType.Propose: {
-                // Synchronously handle quorum ops like regular ops
-                this.handleOp(undefined, op);
-                return;
-            }
-            default: {
-                return;
-            }
-        }
-    }
+    public handleOp(op: ISequencedDocumentMessage) {
+        this.heuristicData.lastOpSequenceNumber = op.sequenceNumber;
```

Comment:
@NicholasCouri Do you think it's dangerous to not adjust `heuristicData.lastOpSequenceNumber` if the incoming op is something we don't care to summarize? Ex: A `Summarize` op comes in and we skip this method entirely and don't update the sequence number.

Reply:
Not sure - if we skip, the time processing those would not be taken in consideration and I wonder if they are neglectable. - not only that but to debug problems it might be interesting to keep the last sequence up-to-date .   @agarwalnavin / @agarwal-navin  might have more background info :)

Reply:
HeuristicData contains info that is relevant to determining whether we want to do a summary or not. Prior to this change, we did not care about most system ops to make a decision on summaries. Looks like with this change, we do consider system ops as well. So, we will have to track them.

This PR seems complicated and I am not really sure what all its changing. Could you add a description and more comments?
From what I can tell, the gist is that it wants to track how many user vs system ops we have received and make summarize decisions based off of that.

Reply:
`lastOpSequenceNumber` is used to know `opsSinceLastAck`. I am not sure what the meaning of that is anymore - do we mean user + system ops since last ack? or just user ops since last ack?

Reply:
`lastOpSequenceNumber` wasn't exactly faithful to its naming in the grand scheme of things. Before, it used to hold the last op sequence number we cared about for summarizing heuristics (i.e. user op batches and the system ops we previously tracked). The system ops we used to care about were `ClientLeave`, `ClientJoin`, and `Propose`. This isn't exactly the best approach as it uses an opt-in mechanism for system ops. This change is trying to better track all the ops that come in before we decide to summarize by providing different weights to different types of ops (user ops = 10x system ops as a base implementation).